server/src/main/java/org/elasticsearch/action/admin/indices/rollover/MetadataRolloverService.java

```diff
@@ -363,6 +410,7 @@ yield new DataStreamAutoShardingEvent(
                 IndexMetadata.builder(newState.metadata().index(originalWriteIndex)).stats(sourceIndexStats).putRolloverInfo(rolloverInfo)
             );
 
+        // TODO: do we need these forecasts for failure stores?
         metadataBuilder = writeLoadForecaster.withWriteLoadForecastForWriteIndex(dataStreamName, metadataBuilder);
         metadataBuilder = withShardSizeForecastForWriteIndex(dataStreamName, metadataBuilder);
```

Comment:
I wasn't sure what the write load forecasters were used for, but it looks like they're used for shard allocation, so makes sense to include them for failure stores as well 👍🏻.

Reply:
I think if we don't predict any harmful effects on regular execution, we should indeed include the forecasts. I suspect we'll need to eventually support some capacity for autosharding on failure indices, and I think this will be needed to support that.